### ISSUE TITLE
feature-specific deps installed inside `pixi shell` only available after restarting shell 

### ISSUE THREAD

**Issue by moritzwilksch** [prefix-dev/pixi#1271]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

(running this in a `fish` shell)

```shell
➜  example git:(main) ✗ pixi init && pixi add python
➜  example git:(main) ✗ pixi add -f feat1 polars
➜  example git:(main) ✗ nvim pixi.toml  # add env1 = ["feat1"]
➜  example git:(main) ✗ pixi shell -e env1
(example:env1) ➜  example git:(main) ✗ python
Python 3.12.3 | packaged by conda-forge | (main, Apr 15 2024, 18:35:20) [Clang 16.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import polars  # works
(example:env1) ➜  example git:(main) ✗ pixi add -f feat1 pandas  # still in the pixi shell
✔ Added pandas
(example:env1) ➜  example git:(main) ✗ python
Python 3.12.3 | packaged by conda-forge | (main, Apr 15 2024, 18:35:20) [Clang 16.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pandas
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'pandas'
>>>
(example:env1) ➜  example git:(main) ✗  # CTRL+D
➜  example git:(main) ✗ pixi shell -e env1
(example:env1) ➜  example git:(main) ✗ python
Python 3.12.3 | packaged by conda-forge | (main, Apr 15 2024, 18:35:20) [Clang 16.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pandas  # works
```

### Issue description

When adding dependencies *to a specific feature* from inside a pixi shell running in an environment using that feature, the installed dependency is not available. One has to quit the pixi shell and restart it for it to be usable.

This does **not** happen when adding deps to the default environment from inside the default shell.

### Expected behavior

Just like with the default environment, dependencies should be available immediately after a `pixi add` without having to restart the pixi shell.

**Comment by niemiaszek** [comment]
Ah, I guess it is what got me confused lately. Same for me

**Comment by ruben-arts** [comment]
Closed in favor of #1018
